core/string/ustring.cpp

```diff
@@ -3243,46 +3243,22 @@ String String::substr(int p_from, int p_chars) const {
 
 int String::find(const String &p_str, int p_from) const {
 	if (p_from < 0) {
-		return -1;
+		p_from = length() + p_from;
 	}
-
-	const int src_len = p_str.length();
-
-	const int len = length();
-
-	if (src_len == 0 || len == 0) {
-		return -1; // won't find anything!
+	if (p_from < 0 || p_from >= length()) {
```

Comment:
Needs to account for large starting indices:

```suggestion
	if (p_from < 0 || p_from + p_str.length() > length() || p_str.is_empty()) {
```

With this, the next `if` block should be deletable entirely, if I'm not mistaken.
Same for `rfind` implementation.